src/tests/multihost/admultidomain/test_multiforest.py

```diff
@@ -0,0 +1,157 @@
+        change_context = 'chcon -t krb5_keytab_t /etc/krb5.keytab-domain1'
+        backup_krb5 = 'cp -rf /etc/krb5.conf /etc/krb5.conf.bak'
+        restore_krb5 = 'mv /etc/krb5.conf.bak /etc/krb5.conf ; ' \
+                       'restorecon -Rv /etc/krb5.conf'
+        cleanup_krb5 = 'rm -rf /etc/krb5.keytab-domain1'
+        edit_krb5_conf = f'sed -i "/domain_realm/a .{ad1_domain} ' \
+                         f'= {ad1_domain_upper}" /etc/krb5.conf'
+        edit1_krb5_conf = f'sed -i "/domain_realm/a {ad1_domain} ' \
+                          f'= {ad1_domain_upper}" /etc/krb5.conf'
+
+        try:
+            multihost.client[0].run_command(get_keytab,
+                                            stdin_text=ad1_password)
+        except subprocess.CalledProcessError:
+            pytest.fail("adcli join failed")
+        multihost.client[0].run_command(backup_krb5, raiseonerr=False)
+        multihost.client[0].run_command(edit_krb5_conf, raiseonerr=False)
+        multihost.client[0].run_command(edit1_krb5_conf, raiseonerr=False)
+        multihost.client[0].run_command(change_context, raiseonerr=False)
+
+        # Configure sssd
+        multihost.client[0].service_sssd('stop')
+        client = sssdTools(multihost.client[0], multihost.ad[0])
+        client.backup_sssd_conf()
+        client.sssd_conf("sssd", {
+            'domains': f'{ad_domain}, {ad1_domain}'}, action='update')
+        domain_params = {
+            'ad_domain': f'{ad_domain}',
+            'dns_discovery_domain': f'{ad_domain}',
+            'ad_server': f'{ad_server}',
+            'debug_level': '9',
+            'use_fully_qualified_names': 'True',
+            'cache_credentials': 'True',
+            'dyndns_update': 'True'}
+        client.sssd_conf(
+            f'domain/{ad_domain}', domain_params, action='update')
+        domain1_params = {
+            'ad_domain': f'{ad1_domain}',
+            'ad_server': f'{ad1_server}',
+            'krb5_realm': f'{ad1_domain_upper}',
+            'debug_level': '9',
+            'use_fully_qualified_names': 'False',
+            'cache_credentials': 'True',
+            'realmd_tags': 'manages-system joined-with-samba',
+            'dyndns_update': 'False',
+            'krb5_keytab': '/etc/krb5.keytab-domain1',
+            'ldap_krb5_keytab': '/etc/krb5.keytab-domain1',
+            'id_provider': 'ad',
+            'access_provider': 'ad',
+            'timeout': '3600',
+            'krb5_store_password_if_offline': 'True',
+            'default_shell': '/bin/bash',
+            'ldap_id_mapping': 'True'}
+        client.sssd_conf(
+            f'domain/{ad1_domain}', domain1_params, action='update')
+        client.clear_sssd_cache()
+        multihost.client[0].service_sssd('start')
+        time.sleep(10)
+        # Search for the user in same forest and domain
+        getent_domain_user1 = multihost.client[0].run_command(
+            f'getent passwd user1@{ad_domain}', raiseonerr=False)
```

Comment:
Is restoring the backups the main reason for running the checks with raiseonerr=False and then doing asserts after the restores?   If so, would it make sense to define a small fixture for this to handle the backups and restores and then you should be able to drop the separate asserts.